New error/success notification

sirius651 opened the pull request:
**Pull Request Summary**

> New Error/success notification

![스크린샷 2023-01-03 오전 11 12 34](https://user-images.githubusercontent.com/1606820/210292067-ce0e3da1-153f-4db4-8e75-0ce26df47f5a.png)

![스크린샷 2023-01-03 오전 11 12 40](https://user-images.githubusercontent.com/1606820/210292070-f8f744ca-47fe-4eb8-9198-f999c66606b5.png)


**Check list**

- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**Release notes:**
feat: added new error/success notification

sirius651 commented:
@Kahonnohak please review for design-wise first. It was temporarily implemented so that three notifications could be seen when entering the screen.

Kahonnohak commented:
Thank you @sirius651 It is very helpful to be able to see all at once:)

The position
Please have 40px margin on the right for desktop header designs. And please place it in the centre with the mobile header.

Other changes
1 - Please change the width to 340px.
2- The text size, I'm sorry but make it 14px please.
3 - Can we have a button with a 1px border before hovering? I know it is a new button but here it does not need to be so strong. 
4- Dark Theme Background (Blue: 001A33, Yellow: 302502, Red: 331106)
5- Light Theme Background (Blue: E5F2FF, Yellow: FDF8E6, Red: FFEEE8)
These colour I would like to use these for all our warning messages and would be good to have these in Astar UI.
6 - Box-shadow (dark: 4px 4px 4px 0px rgb(0 0 0 / 25%), (light: 4px 4px 4px 0px rgb(0 0 0 / 12%)
7 - Could you make the close button invisible and appear when hovering over the toaster and remove the button hover blue colour, please? 

For all events where the toaster pop up, is there a list for messages? Some are at the moment just showing smart contract (dApp Staking but would like to show the dApp name instead. 

Thank you!

sirius651 commented:
@Kahonnohak 
I believe it's updated on feedback now. please check again. 

> For all events where the toaster pop up, is there a list for messages? Some are at the moment just showing smart contract (dApp Staking but would like to show the dApp name instead.
> 
does that mean we should create message history page separately? to implement, we should store the list of the message to browser(like local storage)

Kahonnohak commented:
@sirius651 Thank you!!

In terms of the message - I did not think so but let me explain.
When we stake it will show only a smart-contract but can we say like things below?

`You have successfully nominated Tofu NFT and staked 100SDN.
`
<img width="569" alt="スクリーンショット 2023-01-04 12 24 56" src="https://user-images.githubusercontent.com/77480847/210555020-06632587-8db8-4ad0-b66e-1c96af6b5bd0.png">

sirius651 commented:
> In terms of the message - I did not think so but let me explain. When we stake it will show only a smart-contract but can we say like things below?

Of course, it's possible. the alert message can be customizable.

Kahonnohak commented:
Okay great!

Design is almost done,

1) as I changed the font size, could you also change the line height to 20px as well? It looks a little tight for the eyes.
2) Close icon - the dark theme ones are good, no need for colour change. If you could make it the same for the light theme so we don't mix the colour like blue on red.
3) the hush goes out of the area

<img width="232" alt="スクリーンショット 2023-01-04 16 01 57" src="https://user-images.githubusercontent.com/77480847/210600437-0c8c843d-6a25-4653-8834-acfc584ad6b1.png">
<img width="369" alt="スクリーンショット 2023-01-04 16 09 08" src="https://user-images.githubusercontent.com/77480847/210600855-e1e3584a-d4c9-484d-ab39-281bd40a6c1f.png">


Behaviour - 
When I claimed my rewards  I got these two.  we usually only get one message "you have successfully claimed XXX tokens" Couldn't test more as I've no unclaimed era left.

Text messages
1) Transfer - At the moment you'd get just hush
- Could you make it to 
`You have sent XXX ASTR to wegdty....segrerfd.
`
2) Stake - 
`You successfully staked XXX ASTR to dAppName`

3) Nominate
`You successfully nominated XXX ASTR to dAppNameA from dAppNameB`

Thanks!

sirius651 commented:
> 
> Text messages
> 
> 1. Transfer - At the moment you'd get just hush
> 
> * Could you make it to
>   `You have sent XXX ASTR to wegdty....segrerfd. `
> 
> 2. Stake -
>    `You successfully staked XXX ASTR to dAppName`
> 3. Nominate
>    `You successfully nominated XXX ASTR to dAppNameA from dAppNameB`
> 
> Thanks!

Updated design. please check again. and regarding text change, I changed some messages for xcm transfer, but for others, I need to work separately under the refactoring task that I'm working on now. because it affects many functions which has params or not

Kahonnohak commented:
Thank you understood!

Kahonnohak reviewed (approved):
Looks awesome!

impelcrypto commented:
https://www.loom.com/share/33b4e63e084342ad8dca068cde5338bb

1. Could you make toast's z-index priority higher than the sidebar?
2. I saw there are 2 toasts popped up even though I clicked the copy button only once.

bobo-k2 commented on src/components/common/Notification/NotificationBar.vue:
```diff
@@ -0,0 +1,241 @@
+
+    const alertTypeTitle = computed(() => {
+      if (props.alertType === AlertType.Success) {
+        return 'Success';
```
Is it possible to translate?

sirius651 replied:
Thanks, fixed in [here](https://github.com/AstarNetwork/astar-apps/pull/644/commits/f6b8488d36e8cca415749aca1f3ecee416ee7320) . regarding transaction button, if no txHash, the button would not be visible now. I'll link proper txHash on the refactor PR.

bobo-k2 reviewed (changes requested):
is "Check your transactions" supposed to be clickable? After claiming tokens I got popups below. Cursor changes when I hover buttons, but nothing happens on click. The same happens for token transfer.
<img width="353" alt="image" src="https://user-images.githubusercontent.com/8452361/211298992-53648d53-fe0c-49a3-b5c7-321e9f707106.png">

Copy address gave me duplicated popups
<img width="355" alt="image" src="https://user-images.githubusercontent.com/8452361/211299384-d0dca6ab-1559-4463-a255-6fd52c623aec.png">

Unable to vertically scroll popups when there are to many of them
<img width="1493" alt="image" src="https://user-images.githubusercontent.com/8452361/211300047-4dc74a19-a423-472b-9573-79176951dafe.png">

Kahonnohak commented:
oh I thought the button was not working because of the staging environment. And thank you for spotting we don't want the pop-up when just copied do we? I'd be happy to remove the toaster from copy action. what do you think?


> is "Check your transactions" supposed to be clickable? After claiming tokens I got popups below. Cursor changes when I hover buttons, but nothing happens on click. The same happens for token transfer. <img alt="image" width="353" src="https://user-images.githubusercontent.com/8452361/211298992-53648d53-fe0c-49a3-b5c7-321e9f707106.png">
> 
> Copy address gave me duplicated popups <img alt="image" width="355" src="https://user-images.githubusercontent.com/8452361/211299384-d0dca6ab-1559-4463-a255-6fd52c623aec.png">
> 
> Unable to vertically scroll popups when there are to many of them <img alt="image" width="1493" src="https://user-images.githubusercontent.com/8452361/211300047-4dc74a19-a423-472b-9573-79176951dafe.png">

impelcrypto commented:
@Kahonnohak 
> we don't want the pop-up when just copied do we?
I think it is nice to have the toast for copy action.

Kahonnohak commented:
Okay in that case, "copied" is not really success so let's simplify it please @sirius651 
<img width="725" alt="スクリーンショット 2023-01-09 12 45 30" src="https://user-images.githubusercontent.com/77480847/211311380-85864aec-9ac9-46f1-a60e-5b2ce4f473f1.png">

sirius651 commented:
> https://www.loom.com/share/33b4e63e084342ad8dca068cde5338bb
> 
> 1. Could you make toast's z-index priority higher than the sidebar?
> 2. I saw there are 2 toasts popped up even though I clicked the copy button only once.

Thanks, fixed in [here](https://github.com/AstarNetwork/astar-apps/pull/644/commits/f6b8488d36e8cca415749aca1f3ecee416ee7320)

sirius651 commented:
> Okay in that case, "copied" is not really success so let's simplify it please @sirius651

Fixed in [here](https://github.com/AstarNetwork/astar-apps/pull/644/commits/f58f85dc33b05ca704b7b5abd85a6c784564d4ee)

Kahonnohak commented:
> > Okay in that case, "copied" is not really success so let's simplify it please @sirius651
> 
> Fixed in [here](https://github.com/AstarNetwork/astar-apps/pull/644/commits/f58f85dc33b05ca704b7b5abd85a6c784564d4ee)


Thank you! but now I see this slide bar ((((;´･ω･`))) (My environment has not been changed)

![Uploading スクリーンショット 2023-01-10 6.29.51.png…]()

sirius651 commented:
> Thank you! but now I see this slide bar ((((;´･ω･`))) (My environment has not been changed)
> 
Is the environment on FF? Fixed overflow-x as hidden

Kahonnohak commented:
Was Brave but I see it fixed this time thank you!

bobo-k2 reviewed (approved):
Great work. LGTM

impelcrypto commented on src/components/common/Notification/NotificationBar.vue:
```diff
@@ -0,0 +1,248 @@
+
+    const { t } = useI18n();
+
+    const alertTypeTitle = computed(() => {
```
just a small thing, let's add a type annotation to `ref` and `computed`

sirius651 replied:
Fixed in [here](https://github.com/AstarNetwork/astar-apps/pull/644/commits/c6d329b5df7c4c973a86ec15db606afe352dc81d)

impelcrypto reviewed (changes requested):
Let's add  a type annotation to `ref` and `computed` variables
